content/docs/2.0/concepts/external-scalers.md

```diff
@@ -0,0 +1,582 @@
+  {
+    var earthquakeCount = await GetEarthQuakeCount(longitude, latitude);
+    if (earthquakeCount > 2) {
+      await responseStream.WriteAsync(new IsActiveResponse
+      {
+        Result = true
+      });
+    }
+
+    await Task.Delay(TimeSpan.FromHours(1));
+  }
+}
+```
+{{< /collapsible >}}
+
+{{< collapsible "Javascript" >}}
+```js
+server.addService(externalScalerProto.externalscaler.ExternalScaler.service, {
+  // ...
+  streamIsActive: (call, callback) => {
+    const longitude = call.request.scalerMetadata.longitude
+    const latitude = call.request.scalerMetadata.latitude
+    if (!longitude || !latitude) {
+      callback({
+        code: grpc.status.INVALID_ARGUMENT,
+        details: 'longitude and latitude must be specified',
+      })
+    } else {
+      const interval = setInterval(() => {
+        getEarthquakeCount((err, count) => {
+          if (err) {
+            console.error(err)
+          } else if (count > 2) {
+            call.write({
+              result: true,
+            })
+          }
+        })
+      }, 1000 * 60 * 60);
+
+      call.on('end', () => {
+        clearInterval(interval)
+      })
+    }
+  }
+})
+```
+{{< /collapsible >}}
+
+<br />
+
+#### 5. Implementing `GetMetricSpec`
+
+`GetMetricSpec` returns the `target` value for [the HPA definition for the scaler](https://kubernetes.io/docs/tasks/run-application/horizontal-pod-autoscale-walkthrough/). This scaler will define a static target of 10, but the threshold value is often specified in the metadata for other scalers.
+
+{{< collapsible "Golang" >}}
+```golang
+func (e *ExternalScaler) GetMetricSpec(context.Context, *pb.ScaledObjectRef) (*pb.GetMetricSpecResponse, error) {
+	return &pb.GetMetricSpecResponse{
+		MetricSpecs: []*pb.MetricSpec{{
+			MetricName: "earthquakeThreshold",
+			TargetSize: 10,
+		}},
+	}, nil
+}
+```
+{{< /collapsible >}}
+
+{{< collapsible "C#" >}}
+```csharp
+public override Task<GetMetricSpecResponse> GetMetricSpec(ScaledObjectRef request, ServerCallContext context)
+{
+  var resp = new GetMetricSpecResponse();
+
+  resp.MetricSpecs.Add(new MetricSpec
+  {
+    MetricName = "earthquakeThreshold",
+    TargetSize = 10
+  });
+
+  return Task.FromResult(resp);
+}
+```
+{{< /collapsible >}}
+
+{{< collapsible "Javascript" >}}
+```js
+server.addService(externalScalerProto.externalscaler.ExternalScaler.service, {
+  // ...
+  getMetricSpec: (call, callback) => {
+    callback(null, {
+      metricSpecs: [{
+        metricName: 'earthquakeThreshold',
+        targetSize: 10,
+      }]
+    })
+  }
+})
+```
+{{< /collapsible >}}
+
+<br />
+
+#### 6. Implementing `GetMetrics`
+
+`GetMetrics` returns the value of the metric refered to from `GetMetricSpec`, in this example it's `earthquakeThreshold`. Refer to the [HPA docs](https://kubernetes.io/docs/tasks/run-application/horizontal-pod-autoscale-walkthrough/) for how HPA calculates replicaCount based on metric value and target value.
```

Comment:
```suggestion
`GetMetrics` returns the value of the metric refered to from `GetMetricSpec`, in this example it's `earthquakeThreshold`.

Refer to the [HPA docs](https://kubernetes.io/docs/tasks/run-application/horizontal-pod-autoscale-walkthrough/) for how HPA calculates replicaCount based on metric value and target value.
```

I would even maybe move the last part underneight the samples as it's less relevant here